pkg/controller/clusterstate/api/resource_info.go

```diff
@@ -84,6 +84,20 @@ func NewResource(rl v1.ResourceList) *Resource {
 	return r
 }
 
+func GetCondition(nodeStatus v1.Node) v1.ConditionStatus {
+	var currentNodeStatus v1.ConditionStatus = v1.ConditionFalse
+	condition := 0
+	for condition < len(nodeStatus.Status.Conditions) {
+		// return status of Ready condition.type only
+		if nodeStatus.Status.Conditions[condition].Type == "Ready" {
```

Comment:
Recommend using defined type instead of string, v1.NodeConditionType.NodeReady.

Reply:
Done, thanks